bridge/contracts/utils/auth/ImmutableALCA.sol

```diff
@@ -0,0 +1,31 @@
+// This file is auto-generated by hardhat generate-immutable-auth-contract task. DO NOT EDIT.
+// SPDX-License-Identifier: MIT-open-group
+pragma solidity ^0.8.16;
+
+import "contracts/utils/DeterministicAddress.sol";
+import "contracts/utils/auth/ImmutableFactory.sol";
+import "contracts/interfaces/IAliceNetFactory.sol";
+
+abstract contract ImmutableALCA is ImmutableFactory {
+    address private immutable _alca;
+    error OnlyALCA(address sender, address expected);
+
+    modifier onlyALCA() {
+        if (msg.sender != _alca) {
+            revert OnlyALCA(msg.sender, _alca);
+        }
+        _;
+    }
+
+    constructor() {
+        _alca = IAliceNetFactory(_factoryAddress()).lookup(_saltForALCA());
+    }
+
+    function _alcaAddress() internal view returns (address) {
+        return _alca;
+    }
+
+    function _saltForALCA() internal pure returns (bytes32) {
+        return 0x414c434100000000000000000000000000000000000000000000000000000000;
```

Comment:
Comment magic number or use constant value to add meaning.

Reply:
This will be changed in the following PR!